deepaas/cmd/execute.py

```diff
@@ -0,0 +1,145 @@
+# under the License.
+
+import collections
+import mimetypes
+import os
+import shutil
+import sys
+
+
+from oslo_config import cfg
+from oslo_log import log
+
+from deepaas.model import loading
+
+cli_opts = [
+    cfg.StrOpt('input_file',
+               short="i",
+               required=True,
+               help="""
+Set local input file to predict.
+"""),
+    cfg.StrOpt('content_type',
+               default='application/json',
+               short='ct',
+               help="""
+Especify the content type of the output file. The available options are
+(image/png, application/json (by default), application/zip).
+"""),
+    cfg.StrOpt('output',
+               short="o",
+               required=True,
+               help="""
+Save the result to a local file.
+"""),
+    cfg.BoolOpt('url',
+                short='u',
+                default=False,
+                help="""
+Run as input file an URL.
+If this option is set to True, we can use the URL
+of an image as an input file.
+"""),
+]
+
+CONF = cfg.CONF
+CONF.register_cli_opts(cli_opts)
+
+LOG = log.getLogger(__name__)
+
+# Loading the model installed
+
+
+def model_name():
+    global MODEL_NAME
+    try:
+        for name, model in loading.get_available_models("v2").items():
+            MODEL_NAME = name
+    except Exception as e:
+        LOG.warning("Error loading models: %s", e)
+
+
+def prediction(input_file, file_type, content_type):
+    model_name()  # Function to know the name of the installed model
+
+    package_name = MODEL_NAME
+    predict_data = __import__(package_name).api.predict_data  # Import function
+    predict_url = __import__(package_name).api.predict_url  # Import function
+
+    UploadedFile = collections.namedtuple("UploadedFile", ("name",
+                                                           "filename",
+                                                           "content_type"))
+
+    file = UploadedFile(name=input_file,
+                        filename=input_file,
+                        content_type='image/png')
+
+    if file_type is True:
+        input_data = {'urls': [input_file], 'accept': content_type}
+        output_pred = predict_url(input_data)
+    else:
+        input_data = {'files': [file], 'accept': content_type}
+        output_pred = predict_data(input_data)
+
+    return (output_pred)
+
+
+def main():
+    cfg.CONF(sys.argv[1:])
+    input_file = CONF.input_file
+    content_type = CONF.content_type
+    file_type = CONF.url
+    output = CONF.output
+
+    output_pred = prediction(input_file, file_type, content_type)
+    print(output_pred)
+    extension = mimetypes.guess_extension(content_type)
+    out_file_name = "out_" + os.path.splitext(os.path.basename(input_file))[0]
+    if extension is None:
+        sys.stderr.write(
+            "ERROR: Content type {} not valid.\n".format(content_type))
+        sys.exit(1)
+    if extension == ".json":
```

Comment:
I do not understand what you are doing here with the output. If there is an output and a valid content-type we can simply save the file without any other checks, right? We can have an arbitrary number of content-types supported, so why limiting things here to "png", "json" and "zip?

Reply:
Hi Álvaro, thanks for answering quickly. The problem I see with the outputs is that depending on the content type that is specified, the output is different. In the case of application/json the output of the predict function is a json with the prediction and what I do is save it to a file. In the case of image/png the output (image and an output.json file) and application/zip (.zip file with the prediction of the image and the output.json file) is saved in the /tmp folder and what I do is copy it to the output directory specified on the command line.